test/unit/custom_spec.js

```diff
@@ -107,3 +107,129 @@ describe("custom canvas rendering", function () {
       .catch(done.fail);
   });
 });
+
+describe("alternate document context", function () {
+  const global = typeof globalThis !== "undefined" ? globalThis : window;
+  const FontFace = global.FontFace;
+
+  const ourFont = font => /g_d\d+_f1/.test(font.family);
+  const ourFontRule = rule =>
+    /^@font-face {font-family:"g_d\d+_f1";src:/.test(rule);
+  const has = (fn, iterable) => !![...iterable].find(fn);
```

Comment:
Both this function name, its implementation, and its usage all seem a slightly unclear; can you please re-factor the checking to be a bit more straightforward and easy to follow?